packages/contractkit/src/wrappers/Validators.ts

```diff
@@ -0,0 +1,145 @@
+    const vgAddresses = await this.contract.methods.getRegisteredValidators().call()
+
+    return Promise.all(vgAddresses.map((addr) => this.getValidator(addr)))
+  }
+
+  async getValidator(address: Address): Promise<Validator> {
+    const res = await this.contract.methods.getValidator(address).call()
+    return {
+      address,
+      id: res[0],
+      name: res[1],
+      url: res[2],
+      publicKey: res[3] as any,
+      affiliation: res[4],
+    }
+  }
+
+  async getRegisteredValidatorGroups(): Promise<ValidatorGroup[]> {
+    const vgAddresses = await this.contract.methods.getRegisteredValidatorGroups().call()
+    return Promise.all(vgAddresses.map((addr) => this.getValidatorGroup(addr)))
+  }
+
+  async getValidatorGroup(address: Address): Promise<ValidatorGroup> {
+    const res = await this.contract.methods.getValidatorGroup(address).call()
+    return { address, id: res[0], name: res[1], url: res[2], members: res[3] }
+  }
+
+  async getValidatorGroupsVotes(): Promise<ValidatorGroupVote[]> {
+    const vgAddresses = await this.contract.methods.getRegisteredValidatorGroups().call()
+    const res = await this.contract.methods.getValidatorGroupVotes().call()
+    const r = zip((a, b) => ({ address: a, votes: Web3.utils.toBN(b) }), res[0], res[1])
+    for (const vgAddress of vgAddresses) {
+      if (!res[0].includes(vgAddress)) {
+        r.push({ address: vgAddress, votes: Web3.utils.toBN(0) })
+      }
+    }
+    return r
+  }
+
+  async getVoteFrom(validatorAddress: Address): Promise<Address | null> {
+    return this.contract.methods.voters(validatorAddress).call()
+  }
+
+  async revokeVote(): Promise<TransactionObject<boolean>> {
+    if (this.kit.defaultAccount == null) {
+      throw new Error(`missing from at new ValdidatorUtils()`)
+    }
+
+    const bondedDeposits = await this.kit.contracts.getBondedDeposits()
+    const votingDetails = await bondedDeposits.getVotingDetails(this.kit.defaultAccount)
+    const votedGroup = await this.getVoteFrom(votingDetails.accountAddress)
+
+    if (votedGroup == null) {
+      throw new Error(`Not current vote for ${this.kit.defaultAccount}`)
+    }
+
+    const { lesser, greater } = await this.findLesserAndGreaterAfterVote(
+      votedGroup,
+      votingDetails.weight.neg()
+    )
+
+    return this.contract.methods.revokeVote(lesser, greater)
+  }
+
+  async vote(validatorGroup: Address): Promise<TransactionObject<boolean>> {
+    if (this.kit.defaultAccount == null) {
+      throw new Error(`missing from at new ValdidatorUtils()`)
+    }
+
+    const bondedDeposits = await this.kit.contracts.getBondedDeposits()
+    const votingDetails = await bondedDeposits.getVotingDetails(this.kit.defaultAccount)
+
+    const { lesser, greater } = await this.findLesserAndGreaterAfterVote(
+      validatorGroup,
+      votingDetails.weight
+    )
+
+    return this.contract.methods.vote(validatorGroup, lesser, greater)
+  }
+
+  private async findLesserAndGreaterAfterVote(
```

Comment:
Probably for a future PR, but I would imagine a more general version of this function will be useful for other places we use `SortedLinkedLists`

Reply:
sure, the moment we add other wrappers this will make sense